dace/codegen/targets/fpga.py

```diff
@@ -842,9 +842,14 @@ def generate_node(self, sdfg, dfg, state_id, node, function_stream,
             getattr(self, method_name)(sdfg, dfg, state_id, node,
                                        function_stream, callsite_stream)
         else:
+            old_codegen = self._cpu_codegen.calling_codegen
+            self._cpu_codegen.calling_codegen = self
+
             self._cpu_codegen.generate_node(sdfg, dfg, state_id, node,
```

Comment:
Could you do this with a callback instead?

Reply:
And get rid of `calling_codegen` entirely, if possible

Reply:
because of nested SDFGs in nested SDFGs, which won't preserve the codegen unless I add a `codegen` argument everywhere

Reply:
Fine, burninate in later PR